cmk/gui/plugins/wato/check_parameters/checkpoint_connections.py

```diff
@@ -10,7 +10,7 @@
     rulespec_registry,
     RulespecGroupCheckParametersApplications,
 )
-from cmk.gui.valuespec import Dictionary, Integer, Transform, Tuple
+from cmk.gui.valuespec import Dictionary, Integer, Percentage, Transform, Tuple
 
 
 def _parameter_valuespec_checkpoint_connections() -> Transform:
```

Comment:
I am not sure if this is the right way to build this valuespec. In its current state, users can configure both absolute and relative thresholds. However, I would say that one usually knows whether you want the first or the latter, right? In that case, I think it would be better to use `CascadingDropdown` here, s.t. users first have to decide on the type of levels and then enter the numbers. Or are there actually scenarios where you want to configure both absolute and relative thresholds at the same time? I think this depends on whether fwConnTableLimit can change frequently from a hard limit something that is adjusted automatically and back, which I don't know.

Reply:
I thought about that for a while as well and talked to our firewall colleagues about what they had seen at customer sites and in practice, the setup is sometimes mixed. So you'll see a few firewalls with fwConnTableLimit configured and others at the same customer where it's not configured, this way you don't need to check each firewall and create a tag for it but can simply set one rule to rule them all, as there is always an absolute fall back value.

Reply:
Hmm, I am not yet convinced. Does it make sense to create a single rule for multiple firewalls with different connections? Would you not have to anyway create multiple, host-specific rules? @thl-cmk do you have an opinion on this?

Reply:
hold on - I'm not sure we're talking about the same thing.
If I understand Jörg correctly, he is concerned about the setting of the thresholds, because at the moment I would allow the user to set both absolute + relative thresholds but then only use the relative thresholds if a connection limit is defined and otherwise fall back to the absolute values, and Jörg thinks it makes more sense to set either absolute OR relative thresholds. Whats your opinion on this Thomas?

While Thomas your point sounds like your concerned about the metrics, but if I'm not mistaken about my own code, I already do keep the absolute metrics no matter if the absolute or relative thresholds are used to determine the service state. (Of course the connections_pct are only added in case of the the relative thresholds, but the absolute connection number is always returned)

Adding more values and only yielding the metrics shouldn't be an issue, I'll look into it once we have the other questions figured out :).

Reply:
@gradecke  you abolutly right, i was only thinking about the releative metrics without realy checking your code. Now i have.
I allways have the absolute values and if i like i can have aditional the relative metrics. Thats perfect for me.

Reply:
> Hmm, I am not yet convinced. Does it make sense to create a single rule for multiple firewalls with different connections? Would you not have to anyway create multiple, host-specific rules? @thl-cmk do you have an opinion on this?

Good question, I had to think about it for a while too.
With the dropdown list it's more clear. And yes, depending on my environment I need anyway more than one rule.
The advantage of the current aproche is that I can have a percentage rule with a configurable fallback. 
With the dropdown list the fallback is fixed (default parameters). So if the CheckPoint admin decides to change the config from fixed to automatic (without telling the CMK admin), I need some kind of warning about this change. 

@radecke
> check each firewall and create a tag for it 
I think, you can here add a dynamic hostlabel for fixed connection table limit and match on it.

Reply:
Concerning the metrics, I think the current state is fine (except for some missing metric definitions, see the other comments). We always produce absolute metric and we also produce relative metrics if we can, that sounds fine to me. I have talked to a colleague concerning the design of the rule. We came to the conclusion that a dropdown where the user has to decide between absolute and relative levels is better, simply because its clearer. This is also connected to the point that we try to design rules where the individual configuration options are independent of each other as much as possible.